lib/req/steps.ex

```diff
@@ -285,6 +285,7 @@ defmodule Req.Steps do
       data = request.options[:json] ->
         %{request | body: Jason.encode_to_iodata!(data)}
         |> Req.Request.put_new_header("content-type", "application/json")
+        |> Req.Request.put_new_header("accept", "application/json, */*;q=0.5")
```

Comment:
please also document that we're setting the accept header. Perhaps something like this:

```markdown
* `:json` - (...) It also sets the `accept` header to `application/json`.
```